packages/transaction-controller/src/TransactionController.ts

```diff
@@ -592,14 +587,26 @@ export class TransactionController extends BaseControllerV1<
 
     this.addPendingTransactionTrackerListeners();
 
+    this.subscribe(this.#onStateChange)
+
     onNetworkStateChange(() => {
       log('Detected network change', this.getChainId());
+      // NOTE(JL): Network state changes also trigger PendingTransactionTracker's onStateChange
```

Comment:
why is this?

Reply:
https://github.com/MetaMask/core/commit/89654542c9c61308cfad6a310f7fe2b4b669117b#diff-e9d3bdb90733c34ccfaaf1038b2098cc5d0486f4cfe8ec127e6335862ac8b93f added being triggered by onNetworkStateChange